charts/cloudzero-agent/docs/cert-trouble-shooting.md

```diff
@@ -0,0 +1,367 @@
+
+1. **Generate Certificates**:
+   Ensure the Common Name is `<RELEASE_NAME>.<RELEASE_NAMESPACE>.cluster.local`.
+
+2. **Create Kubernetes Secret**:
+   ```bash
+   kubectl create secret tls <YOUR_TLS_SECRET_NAME> \
+     --cert=path/to/tls.crt \
+     --key=path/to/tls.key \
+     --namespace <YOUR_NAMESPACE>
+   ```
+
+   Additionally, create a secret for the CA bundle:
+   ```bash
+   kubectl create secret generic <YOUR_CA_SECRET_NAME> \
+     --from-file=ca.crt=path/to/ca.crt \
+     --namespace <YOUR_NAMESPACE>
+   ```
+
+3. **Update Configuration**:
+   Modify `configuration.example.yaml` to reference your custom secrets:
+   ```yaml
+   insightsController:
+     server:
+       tls:
+         nameOverride: <YOUR_TLS_SECRET_NAME>
+     webhooks:
+       caBundle: <BASE64_ENCODED_CA_BUNDLE>
+     certificate:
+       enabled: false
+     issuer:
+       enabled: false
+   cert-manager:
+     enabled: false
+   ```
+
+4. **Reinstall Helm Chart**:
+   ```bash
+   helm upgrade --install <RELEASE_NAME> cloudzero/cloudzero-agent -f configuration.example.yaml
+   ```
+
+### 4. Correct Certificate References in Configuration
+
+Ensure that all certificate-related fields in your configuration files correctly reference the existing secrets and CA bundles.
+
+- **Verify `nameOverride`**:
+  The `nameOverride` under `insightsController.server.tls` should match the name of your TLS secret.
+
+- **Ensure `caBundle` is Correct**:
+  The `caBundle` should contain the base64-encoded CA certificate that signed the webhook server certificate.
+
+- **Disable Conflicting Certificate Managers**:
+  If you switch from one certificate management method to another, ensure that conflicting settings (like `cert-manager.enabled`) are appropriately disabled.
+
+---
+
+## Best Practices
+
+- **Automate Certificate Renewal**: Use `cert-manager` or similar tools to handle certificate issuance and renewal automatically.
+- **Monitor Certificate Expiry**: Implement monitoring to alert you before certificates expire.
+- **Secure Secret Management**: Ensure that Kubernetes Secrets containing certificates are securely managed and access-controlled.
+- **Consistent Naming Conventions**: Maintain consistent naming for secrets and release names to avoid configuration mismatches.
+- **Regularly Update Dependencies**: Keep `cert-manager` and other dependencies up to date to benefit from security patches and improvements.
+
+---
+
+## Additional Resources
+
+- [Kubernetes Admission Controllers Documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+- [`cert-manager` Documentation](https://cert-manager.io/docs/)
+- [CloudZero Agent Helm Chart README](./README.md)
+- [ValidatingWebhookConfiguration Reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#validatingwebhookconfiguration-v1-admissionregistration-k8s-io)
+- [Helm Troubleshooting Guide](https://helm.sh/docs/chart_best_practices/troubleshooting/)
```

Comment:
This URL is broken; should we link to `https://helm.sh/docs/faq/troubleshooting/` instead?